File: java/src/org/gessinger/gepard/ActionCmd.java
package org.gessinger.gepard ;

import java.util.* ;

public class ActionCmd
{
  Map<String,Object> parameter ;
  String cmd ;
  Map<String,String> args ;
  String result = "" ;

  public ActionCmd ( Map<String,Object> parameter )
  {
    this.parameter = parameter ;
    this.cmd       = (String) parameter.get ( "cmd" ) ;
    this.args      = (Map<String,String>) parameter.get ( "args" ) ;
  }
  public String getCmd (  )
  {
    return cmd ;    
  }
  public void setResult ( String str )
  {
    result = str ;
  }
  public Map<String,String> getArgs()
  {
    return args ;
  }
}

File: src/ActionCmd.js
/* 
* @Author: Hans Jürgen Gessinger
* @Date:   2016-01-19 12:50:43
*/

'use strict';

var ActionCmd = function ( cmd )
{
  this.cmd = cmd ;
  this.parameter = {} ;
  this.result = "" ;
};
ActionCmd.prototype =
{
  setResult: function ( text )
  {
    this.result = text ;
  },
  getCmd: function()
  {
    return this.cmd ;
  },
  getArgs: function()
  {
    return this.parameter.args ;
  }
};

module.exports = ActionCmd ;
